OpenRA.Mods.Common/Activities/DockUtils.cs

```diff
@@ -0,0 +1,77 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public static class DockUtils
+	{
+		public static Activity GenericApproachDockActivities(Actor host, Actor client, Dock dock,
+			Activity requester, bool goThroughHost = false)
+		{
+			/* Not yet, this is part 1 of the PR where harvester docking is addressed.
```

Comment:
Mixing multiple different movement types in one activity is not very nice at all.  This would be better done as a new `MoveToDock()` method on the `IMove` interface.